internal/pkg/gateway/registry.go

```diff
@@ -260,68 +260,56 @@ func contains(slice []string, entry string) bool {
 
 // Returns a set of broadcastClients that can order a transaction for the given channel.
 func (reg *registry) orderers(channel string) ([]*orderer, error) {
-	err := reg.registerChannel(channel)
-	if err != nil {
-		return nil, err
-	}
 	var orderers []*orderer
-
-	// Get the config
-	config, err := reg.discovery.Config(channel)
-	if err != nil {
-		return nil, err
+	var ordererEndpoints []*endpointConfig
+	var err error
+	addr, exists := reg.channelOrderers.Load(channel)
```

Comment:
I am not sure you can achieve the task in a lock free manner. 

At some point you will need to disconnect orderers when they are no longer in your config, right? 
When that happens, you will have to lock, otherwise you are risking that a goroutine will load the orderers from the map, and then in parallel the orderer's connection will be closed, and right afterwards, that orderer will be connected to, and you will end up with a connection to an orderer that doesn't exist anymore. 

What do you think?

Reply:
I haven't implemented orderer/peer disconnection yet.  When I do, locking may well have to be added here.  But I'd rather do that as part of that future PR rather than in anticipation of something that hasn't been implemented.